programs/mango-v4/src/state/mango_account.rs

```diff
@@ -2862,4 +2864,89 @@ mod tests {
 
         assert_eq!(to_be_closed_account_opt.unwrap().market_index, 3)
     }
+
+    // Attempts reading old mango account data with borsh and with zerocopy
+    #[test]
+    fn test_mango_account_backwards_compatibility() -> Result<()> {
+        use solana_program_test::{find_file, read_file};
+
+        let mut d = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        d.push("resources/test");
+
+        // Grab live accounts with
+        // solana account CZGf1qbYPaSoabuA1EmdN8W5UHvH5CeXcNZ7RTx65aVQ --output-file programs/mango-v4/resources/test/mangoaccount-v0.21.3.bin
+        let fixtures = vec!["mangoaccount-v0.21.3"];
+
+        for fixture in fixtures {
+            let filename = format!("resources/test/{}.bin", fixture);
+            let account_bytes = read_file(find_file(&filename).unwrap());
+
+            // Try a roundtrip via borsh
+            {
+                let mut account_bytes_slice: &[u8] = &account_bytes;
+                let data = MangoAccount::try_deserialize(&mut account_bytes_slice)?;
+                let mut borsh_bytes = Vec::new();
+                data.try_serialize(&mut borsh_bytes)?;
```

Comment:
do we actually need to serialize back in the same format ?
if we don't, maybe do this instead:

```old_serialized = read_file()
old_deserialized = deserialize(old_serialized)
new_serialized = serialized(old_deserialized)
new_deserialized = deserialize(new_serialized)
assert_eq(old_deserialized, new_deserialized)```